### ISSUE TITLE
Invoke-D365DbSyncModule wildcard

### ISSUE THREAD

**Issue by caleblanchard** [d365collaborative/d365fo.tools#399]
According to the Get-Help, the Invoke-D365DbSyncModule cmdlet should be able to accept wildcards in the module name parameter.

>  -------------------------- EXAMPLE 1 --------------------------
> 
>  PS C:\>Invoke-D365DbSyncModule -Module "Application*Adaptor"
> 
>  Retrieve the list of installed packages / modules where the name fits the search "Application*Adaptor".
> 
>  It will run loop over the list and start the sync process against all tables, views, data entities, table-extensions,
>  view-extensions and data entities-extensions of every iterated model

When using anything other than the exact module name, the cmdlet fails:

> PS C:\windows\system32> Invoke-D365DbSyncModule -Module "Application*Adaptor"
> Invoke-D365DBSyncPartial : Cannot bind argument to parameter 'SyncList' because it is an empty array.
> At line:15195 char:51
> \+         $resSyncModule = Invoke-D365DBSyncPartial @syncParams
> \+                                                   ~~~~~~~~~~~
>     \+ CategoryInfo          : InvalidData: (:) [Invoke-D365DBSyncPartial], ParameterBindingValidationException
>     \+ FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyArrayNotAllowed,Invoke-D365DBSyncPartial

This appears to be because the MetadataProvider class used in the internal function Get-SyncElements does not support wildcards in the module names (I decompiled the code in that DLL and confirmed that it does not (10.0.8 PU32). It expects the exact modulename).

I propose that instead of wildcards, a comma separated list like Clear-D365TableDataFromBacpac does with the TableName parameter should be supported.

I am happy to make this change if it is agreed!

**Comment by Splaxi** [comment]
Thanks for providing feedback about this @caleblanchard!

I have to admit that I didn't implement this myself, so I might explain why the get-help content and the real world isn't fully aligned.

I propose 2 things:
1. We should extend the parameter attribute for the `-Module` / `-ModuleName` 

```powershell
        [Parameter(Mandatory = $true, ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)
        [Alias("ModuleName")]
        [string] $Module
``` 
This will enable us to utilize the `Get-D365Module` which in fact supports wildcards, and then you can pipe the output like this:

```powershell
Get-D365Module -Name "Application*Adaptor" | Invoke-D365DbSyncModule
```

We (you) need to test that it actually produces what we (you) are after. When that is done, we can continue with number 2.

2. We should most definitely add support for a list of modules/modulenames directly in the parameter, like this:

```powershell
        [Parameter(Mandatory = $true, ValueFromPipeline = $true, ValueFromPipelineByPropertyName = $true)
        [Alias("ModuleName")]
        [string[]] $Module
``` 
 
But requires us (you) to implement a foreach / for loop pattern in the `Invoke-D365DbSyncModule` and test that both scenarios work.

Then we should be on top of things and actually support different ways of using the cmdlet / function. I will update the documentation afterwards, to be aligned with what the cmdlet / function is capable of. I would appreciate if you could share your different test scripts / commands, so I can rewrite them into some generic, and show people how we know it will work based on our implementation.

How does that sound to you?

**Comment by caleblanchard** [comment]
I agree. That would be the best way forward. I will start on the implementation.

**Comment by Splaxi** [comment]
Awesome!

Let me know if you need anything 😀

**Comment by Splaxi** [comment]
Should be fixed in 0.5.82...
